src/data_structures.rs

```diff
@@ -59,68 +88,59 @@ pub trait PCVerifierKey:
     fn max_degree(&self) -> usize;
 
     /// Outputs the maximum degree supported by the verifier key.
-    fn supported_degree(&self) -> usize;
+    fn segment_size(&self) -> usize;
 
     /// Returns the hash of `self` instance.
     fn get_hash(&self) -> &[u8];
 
     /// Randomize key for testing purpose
-    fn randomize(&mut self);
+    // #[cfg(test)]
+    fn ut_randomize(&mut self);
 }
 
-/// Defines the minimal interface of prepared verifier keys for any polynomial
-/// commitment scheme.
-pub trait PCPreparedVerifierKey<Unprepared: PCVerifierKey> {
-    /// prepare
-    fn prepare(vk: &Unprepared) -> Self;
+/// Defines the minimal interface of the verifier state from succinct
+/// verify part.
+/// This state is returned by the succinct verifier and contains everything
+/// that is needed to complete verification.
+pub trait PCVerifierState: Clone + Debug + Eq + PartialEq {}
+
+/// Defines the minimal interface for opening proofs for a polynomial commitment
+pub trait PCProof:
+    Clone + Debug + Eq + PartialEq + CanonicalSerialize + CanonicalDeserialize + SemanticallyValid
+{
+    /// Return length of committer key
+    fn get_key_len(&self) -> usize;
 }
 
-/// Defines the minimal interface of commitments for any polynomial
-/// commitment scheme.
-pub trait PCCommitment:
-    Clone + CanonicalSerialize + CanonicalDeserialize + ToBytes + SemanticallyValid
+/// Defines the minimal interface for batch evaluation proofs in the style
```

Comment:
Need help from @UlrichHaboeck75 here to answer. From code point of view both are possible: rename it to something more generic or convert this to struct with more specific name

Reply:
I was thinking the same when reading the code. Currently, `PCMultipointProof` is assumed to be the exactly the Boneh one, as produced by `multi_point_multi_poly_open()`. Since I do not think we ever face the situation of two different types of multi-point proofs (for linear schemes),  let us just rename it. How about `BDFGMultiPointProof`?

Reply:
Fine for me !